### ISSUE TITLE
Interest in async interface?

### ISSUE THREAD

**Issue by gerw** [Bouni/python-luxtronik#133]
Out of curiosity, I tried to implement an asynchronous version of the socket connection to the heat pump, see https://github.com/gerw/python-luxtronik/tree/async. (Beware: this is not polished and, e.g., lacks some basic error handling)

Is there any interest in adding an async version to the main branch? I have seen that HomeAssistant is async-based, would it be beneficial for the integration?

**Comment by kbabioch** [comment]
Personally I think this might be a good improvement. While I'm not super experienced with `asyncio` in Python, an asynchronous computing model makes a lot of sense when dealing with sockets, etc.

So I would definitely vote with "yes" here. On the other hand it might introduce some complexity and will make it harder to debug / maintain / contribute / test. So let's see what the others think about it.

**Comment by kbabioch** [comment]
What I'm missing in general (and that is not related to this feature necessarily) is a roadmap / milestones. Currently we just implement new features and introduce (breaking) changes all over the place with no particular order / plan.

Ideally we should come up with a plan / strategy, i.e. for milestone A we want to have X, Y, Z. For milestone A+1 (=B) we can then focus on XX, YY, ZZ, etc. But let's discuss this at another place.

**Comment by Guzz-T** [comment]
I'm not familiar with `await` and `asyncio`, but shouldn't it be enough to add an `await` to the uses of `socket`?

Allowing a break before each function call would, in my opinion, increase execution time rather than decrease it. After all, the CPU registers are overwritten with every thread change.

**Comment by kbabioch** [comment]
> Allowing a break before each function call would, in my opinion, increase execution time rather than decrease it. After all, the CPU registers are overwritten with every thread change.

I don't think that this is what we need to worry about (at this point in time). (Technically there is hyper threading, etx.), so switching contexts is not expensive at all. Also when dealing with network sockets, register timing is really not critical, since it is orders magnitudes faster.

The main advantage of asynchronous computation is that you can process data as it comes in and don't have to busy wait for it.

Maybe we should summarize what our current problem is that we're trying to solve here and only introduce the additional complexity once it is needed.

**Comment by rhammen** [comment]
I'd like to pick this back up with a concrete, working piece of it.

**The problem, concretely**: I work on the [Luxtronik Home Assistant custom integration](https://github.com/BenPru/luxtronik/), and it currently has to wrap every call into this library with `hass.async_add_executor_job(...)` (a thread-pool shim) because the library itself is fully synchronous. That's extra latency, an extra thread per call, and a chunk of integration code that only exists to route around blocking I/O. Native async support here would let the integration call the library directly.

I looked at @gerw's [`aluxtronik.py` prototype](https://github.com/gerw/python-luxtronik/blob/async/luxtronik/aluxtronik.py) linked above — the core approach (`asyncio.open_connection` + `StreamReader`/`StreamWriter` + `asyncio.Lock`) is exactly right and matches what I ended up with independently. It predates the `cfi`/`shi` split though, so it doesn't map onto current `main` directly.

**Proposal — convert this library to Async, split into two parts, in order:**

1. **Config interface (port 8889) first.** This is the part I can actually test against my own heat pump, so it's the one I want to get right and prove out before touching anything else. I have a branch with this done:
   - https://github.com/rhammen/python-luxtronik/pull/1 (branch `async-cfi-rewrite`)
   - `LuxtronikSocketInterface`'s read/write methods are now coroutines
   - Connection handling moved from "open and close a socket every call" to a persistent connection with automatic reconnect and retry-with-backoff — this mirrors a hardened version of the CFI client I've been running in production in my HA integration for a while, which turned out to handle real-world flakiness better than the original per-call-reconnect approach
   - `connect()`/`close()` plus `async with` support for lifecycle management
   - Full test suite converted and green, and I've verified it live against my own heat pump (read parameters/calculations/visibilities, persistent-connection reuse, and reconnect-after-disconnect all confirmed working)
   - It's a breaking API change (sync → async), which given this is pre-1.0 and marked `[Unreleased]` in the changelog, seemed like the right time to make it

2. **Modbus/SHI second, once (1) is accepted.** I don't have Modbus-capable hardware myself, so I can't verify that part live the way I could the CFI part — I'd want to get the config-interface approach validated first, and would need help from someone who does have SHI-capable hardware to properly test the Modbus follow-up before it merges.

Full disclosure: I worked through this implementation with AI assistance (Claude/Claude Code), including the architecture decisions above — I'm not trying to pass it off as manual work. Happy to walk through any part of the reasoning or change anything the reviewers want done differently.

Would appreciate any early feedback on the PR before I put more time into the Modbus half.
